src/CSharp/CodeCracker/Refactoring/XmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxKind;
+using System;
+using System.Collections.Generic;
+
+namespace CodeCracker.CSharp.Refactoring
+{
+    public abstract class XmlDocumentationCodeFixProvider : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DiagnosticId.XmlDocumentation.ToDiagnosticId());
+
+        public abstract SyntaxNode FixParameters(MethodDeclarationSyntax method, SyntaxNode root);
+
+        public abstract string MessageFormat { get; }
+
+        protected async Task<Document> FixParametersAsync(Document document, Diagnostic diagnostic, CancellationToken c)
+        {
+            var root = await document.GetSyntaxRootAsync(c).ConfigureAwait(false);
+            var documentationNode = (MethodDeclarationSyntax)root.FindNode(diagnostic.Location.SourceSpan);
+            var newRoot = FixParameters(documentationNode, root);
+            var newDocument = document.WithSyntaxRoot(newRoot);
+
+            return newDocument;
+        }
+
+        protected static IEnumerable<Tuple<ParameterSyntax, Tuple<XmlElementSyntax, XmlNameAttributeSyntax>>> GetMethodParametersWithDocParameters(MethodDeclarationSyntax method, DocumentationCommentTriviaSyntax documentationNode)
+        {
+            var methodParameters = method.ParameterList.Parameters;
+
+            var xElementsWitAttrs = documentationNode.Content.OfType<XmlElementSyntax>()
+                                    .Where(xEle => xEle.StartTag.Name.LocalName.ValueText == "param")
+                                    .SelectMany(xEle => xEle.StartTag.Attributes, (xEle, attr) => new Tuple<XmlElementSyntax, XmlNameAttributeSyntax>(xEle, (XmlNameAttributeSyntax)attr));
+
+            var methodParameterLookUp = methodParameters.ToLookup(parameter => parameter.Identifier.ValueText);
+            var docParameterLookUp = xElementsWitAttrs.ToLookup(x => x.Item2.Identifier.Identifier.ValueText);
+
+            var keys = methodParameterLookUp.Select(p => p.Key).Union(docParameterLookUp.Select(p => p.Key)).ToImmutableHashSet();
```

Comment:
It does not seem necessary to create 2 lookups if the only reason for their existence is to later create a union on their keys. Analyzers are all about performance, simply create a hash from the parameter identifiers and xelements without creating the intermediate lookup.